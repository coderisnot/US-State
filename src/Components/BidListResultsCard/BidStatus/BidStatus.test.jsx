import { shallow } from 'enzyme';
import React from 'react';
import toJSON from 'enzyme-to-json';
import BidStatus from './BidStatus';
import { APPROVED, CLOSED, DRAFT, DECLINED, HAND_SHAKE_ACCEPTED, HAND_SHAKE_OFFERED,
HAND_SHAKE_DECLINED, IN_PANEL, SUBMITTED } from '../../../Constants/BidStatuses';

describe('BidStatusComponent', () => {
  it('is defined', () => {
    const wrapper = shallow(
      <BidStatus
        status={APPROVED.property}
      />,
    );
    expect(wrapper).toBeDefined();
  });

  it('matches snapshot for APPROVED', () => {
    const wrapper = shallow(
      <BidStatus
        status={APPROVED.property}
      />,
    );
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('matches snapshot for CLOSED', () => {
    const wrapper = shallow(
      <BidStatus
        status={CLOSED.property}
      />,
    );
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('matches snapshot for DRAFT', () => {
    const wrapper = shallow(
      <BidStatus
        status={DRAFT.property}
      />,
    );
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('matches snapshot for DECLINED', () => {
    const wrapper = shallow(
      <BidStatus
        status={DECLINED.property}
      />,
    );
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('matches snapshot for HAND_SHAKE_ACCEPTED', () => {
    const wrapper = shallow(
      <BidStatus
        status={HAND_SHAKE_ACCEPTED.property}
      />,
    );
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('matches snapshot for HAND_SHAKE_OFFERED', () => {
    const wrapper = shallow(
      <BidStatus
        status={HAND_SHAKE_OFFERED.property}
      />,
    );
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('matches snapshot for HAND_SHAKE_DECLINED', () => {
    const wrapper = shallow(
      <BidStatus
        status={HAND_SHAKE_DECLINED.property}
      />,
    );
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('matches snapshot for IN_PANEL', () => {
    const wrapper = shallow(
      <BidStatus
        status={IN_PANEL.property}
      />,
    );
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('matches snapshot for SUBMITTED', () => {
    const wrapper = shallow(
      <BidStatus
        status={SUBMITTED.property}
      />,
    );
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('matches snapshot for other statuses', () => {
    const wrapper = shallow(
      <BidStatus
        status="other"
      />,
    );
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
