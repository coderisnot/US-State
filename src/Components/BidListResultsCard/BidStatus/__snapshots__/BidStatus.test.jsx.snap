// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BidStatusComponent matches snapshot for APPROVED 1`] = `
<div
  className="bid-list-card-title-status"
>
  <FontAwesome
    name="check"
  />
   
  Approved
</div>
`;

exports[`BidStatusComponent matches snapshot for CLOSED 1`] = `
<div
  className="bid-list-card-title-status"
>
  <FontAwesome
    name="times-circle-o"
  />
   
  Closed
</div>
`;

exports[`BidStatusComponent matches snapshot for DECLINED 1`] = `
<div
  className="bid-list-card-title-status"
>
  <FontAwesome
    name="times"
  />
   
  Bid Unapproved
</div>
`;

exports[`BidStatusComponent matches snapshot for DRAFT 1`] = `
<div
  className="bid-list-card-title-status"
>
  <FontAwesome
    name="bookmark"
  />
   
  Draft Bid
</div>
`;

exports[`BidStatusComponent matches snapshot for HAND_SHAKE_ACCEPTED 1`] = `
<div
  className="bid-list-card-title-status"
>
  <FontAwesome
    name="handshake-o"
  />
   
  Handshake Accepted
</div>
`;

exports[`BidStatusComponent matches snapshot for HAND_SHAKE_DECLINED 1`] = `
<div
  className="bid-list-card-title-status"
>
  <FontAwesome
    name="times"
  />
   
  Handshake Declined
</div>
`;

exports[`BidStatusComponent matches snapshot for HAND_SHAKE_OFFERED 1`] = `
<div
  className="bid-list-card-title-status"
>
  <FontAwesome
    name="star"
  />
   
  Handshake Offered
</div>
`;

exports[`BidStatusComponent matches snapshot for IN_PANEL 1`] = `
<div
  className="bid-list-card-title-status"
>
  <FontAwesome
    name="clock-o"
  />
   
  In Panel
</div>
`;

exports[`BidStatusComponent matches snapshot for SUBMITTED 1`] = `
<div
  className="bid-list-card-title-status"
>
  <FontAwesome
    name="thumbs-o-up"
  />
   
  Bid Submitted
</div>
`;

exports[`BidStatusComponent matches snapshot for other statuses 1`] = `
<div
  className="bid-list-card-title-status"
>
  <FontAwesome
    name="question"
  />
   
  Status unknown
</div>
`;
